File: src/DNASequenceHybridization.java
import java.util.Scanner;

class DNASequenceHybridization {

    private Scanner system = new Scanner(System.in);

    void start() {
        System.out.println("Please type in how long DNA you want to be: ");
        int lenght = system.nextInt();
        String sequence = DNAUtil.createDNASequence(lenght);
        String[] codons = DNAUtil.createCodonsFrom(sequence);
        System.out.println("Generated sequence is : " + sequence);
        printCodons(codons);

    }

    private void printCodons(String[] codons) {
        StringBuilder output = new StringBuilder();

        for (int i = 0; i < codons.length; i++) {
            output.append(codons[i]);
            if (i != codons.length - 1) {
                output.append(", ");
            }
        }

        System.out.println("Codons are : { " + output + " }");
    }
}

File: src/DNASequenceHybridization.kt
import java.util.*

class DNASequenceHybridization {

    private val scanner: Scanner = Scanner(System.`in`)

    fun start() {
        System.out.println("Podaj długość k : ")
        val k: Int = scanner.nextInt()
        PermutationCreator.printAllKLength(k)
        val permutationsList = PermutationCreator.possibleNucleotyde

        System.out.println("Podaj liczbę słów o długości k :")
        val n: Int = scanner.nextInt()

        var errorCounter = 0
        val listOfSequences = mutableListOf<String>()

        for (i in 0..n) {
            val sequence= scanner.nextLine().trim()

            when {
                sequence.length != k -> errorCounter++
                permutationsList.contains(sequence).not() -> errorCounter++
                listOfSequences.contains(sequence) -> errorCounter++
                else -> listOfSequences.add(sequence)
            }

            if (errorCounter > 3) {
                System.out.println("Liczba błędów użytkownika większa niż 3. Uruchom program ponownie!")
                return
            }
        }

        val shortestSuperStringUtil = ShortestSuperStringUtil()
        val superString = shortestSuperStringUtil.createSuperString(listOfSequences)
        System.out.println("Spectrum to : $superString")

    }
}
